server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -591,24 +591,20 @@ private static void parseArray(DocumentParserContext context, String lastFieldNa
 
     private static void parseArrayDynamic(DocumentParserContext context, String currentFieldName) throws IOException {
         ensureNotStrict(context, currentFieldName);
-        if (context.dynamic() == ObjectMapper.Dynamic.FALSE) {
```

Comment:
This is the removed part, the rest is just shifted.

Reply:
If I run the `DynamicMappingTests#testArraysDynamicFalse()` test then the if statement does seem to be executed.

Reply:
The test still passes though, I wonder if it has any effect in practice.

Reply:
So what happens is that the function calls #parseDynamicValue that also checks for this, returning immediately.

We can remove it to simplify the logic, but it's fine to keep it too.

Reply:
I think the test doesn't validate that `skipChildren()` is invoked, which I think is the point of the whole if statement (avoid parsing stuff that we're not going to use). Not sure how, but I think the test should somehow assert that `skipChildren()` is invoked.

Reply:
This is actually needed, I added more tests.. lemme revert for now.